mypy/stubtest.py

```diff
@@ -281,6 +282,34 @@ def _verify_exported_names(
     )
 
 
+def _get_imported_symbol_names(runtime: types.ModuleType) -> frozenset[str] | None:
+    """Retrieve the names in the global namespace which are known to be imported.
+
+    1). Use inspect to retrieve the source code of the module
+    2). Use symtable to parse the source and retrieve names that are known to be imported
+        from other modules.
+
+    If either of the above steps fails, return `None`.
+
+    Note that if a set of names is returned,
+    it won't include names imported via `from foo import *` imports.
+    """
+    try:
+        source = inspect.getsource(runtime)
+    except (OSError, TypeError, SyntaxError):
+        return None
+
+    if not source.strip():
+        return None
```

Comment:
When does inspect.getsource return an empty str? Should this be `return frozenset()`?

Reply:
I think you're right. `inspect.getsource()` _should_ only return an empty string if the source code is an empty file (e.g. if it's an empty `__init__.py` file). And in that case, we know that there'll be 0 imported modules. So returning `frozenset()` is the best solution here.